inc/tracking/class-tracking.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Pressbooks\Tracking;
+
+abstract class Tracking {
+	protected static $instance;
+
+	/**
+	 * Tracking table
+	 *
+	 * @var string
+	 */
+	protected $dbTable;
+
+	/**
+	 * Tracking type
+	 *
+	 * @var string
+	 */
+	protected $type;
+
+	protected function __construct() {
+		global $wpdb;
+
+		$this->dbTable = $wpdb->base_prefix . 'pressbooks_tracking';
+	}
+
+	public static function init() {
+		if ( is_null( static::$instance ) ) {
+			static::$instance = new static;
+
+			static::$instance->setup();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Set up the database table.
+	 *
+	 * @return void
+	 */
+	protected function setup() {
+		require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+
+		$sql = "CREATE TABLE IF NOT EXISTS `$this->dbTable` (
```

Comment:
I think we could store the timezone as well, since some networks is in different timezone. So, when we want to get data from different networks, we can process the data in a single timezone. I see in line 69 we are storing data in the server timezone (which makes sense), but timezone that indicates the in which timezone was the date stored would be good. What do you think?

Reply:
I see no problem in storing the timezone as well, just not sure about use cases for that field. Maybe we could ask @SteelWagstaff ?

Reply:
I am agree, let's ask and we can decide after that.
I was thinking if we want to consume the data across networks in retool for example, if you see in the ticket Steel mentioned:
> As a product owner, I want to know which file types are most frequently downloaded across all networks

Which means if I want to see all books downloaded this week (or today) -for example- for all networks, could be an issue, since networks are in different timezone, the same issue we had with the lastUpdated filter feature in the PB Directory app. 
If we store the timezone, we will be able to get the info avoiding this issue by displacing the time with a centralized timezone (i.e UTC 0)

Reply:
Solved by considering our [second option](https://github.com/pressbooks/pressbooks/pull/2204#discussion_r649183895)